cmd/kubectl-k8ssandra/register/register_test.go

```diff
@@ -59,55 +70,28 @@ func TestRegister(t *testing.T) {
 		SourceNamespace:  "source-namespace",
 		DestNamespace:    "dest-namespace",
 		ServiceAccount:   "k8ssandra-operator",
-		Context:          context.TODO(),
+		Context:          ctx,
 		DestinationName:  "test-destination",
 	}
-	ctx := context.Background()
-
+	// Continue reconciliation
 	require.Eventually(func() bool {
-		if err := ex.RegisterCluster(); err != nil {
-			if errors.Is(err, NonRecoverableError{}) {
-				require.FailNow(fmt.Sprintf("Registration failed: %s", err.Error()))
-			}
-			if err.Error() == "no secret found for service account k8ssandra-operator" {
-				return true
-			}
-			return false
-		}
-		return true
-	}, time.Second*5, time.Millisecond*100)
-
-	// This relies on a controller that is not running in the envtest.
+		res := ex.RegisterCluster()
+		return res == nil
+	}, time.Second*300, time.Second*1)
```

Comment:
issue: Reduce the polling interval (and the maximum time so that we bailout faster). I reduced these already to speed up the test from 16s to 2.4s and I'd hate to see the test taking more time than necessary.